app/features/welcome/components/Welcome.js

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React, { Component } from 'react';
+import URL from 'url';
+
+// $FlowFixMe
+import styled from 'styled-components';
+
+const CenterContent = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+`;
+
+const WelcomeWrapper = CenterContent.extend`
+    background: linear-gradient(#165ecc,#44A5FF);
+    height: 100vh;
+    width: 100%;
+`;
+
+// $FlowFixMe
+import { FieldTextStateless } from '@atlaskit/field-text';
+
+// $FlowFixMe
+import Button from '@atlaskit/button';
+
+// $FlowFixMe
+import { AtlasKitThemeProvider } from '@atlaskit/theme';
+
+
+type Props = {
+
+    /**
+     * React Router history object (for flow)
+     */
+    history: Object;
+};
+
+type State = {
+
+    /**
+     * URL of the room to join
+     * If this is not a url it will be treated as room name for default domain
+     */
+    url: string;
+};
+
+/**
+ * Welcome Component.
+ */
+export default class Welcome extends Component<Props, State> {
+
+    /**
+     * Initialize Welcome Component
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            url: ''
+        };
+
+        // Bind event handlers
+
+        // $FlowFixMe
+        this._onURLChange = this._onURLChange.bind(this);
+
+        // $FlowFixMe
+        this._onFormSubmit = this._onFormSubmit.bind(this);
+
+        // $FlowFixMe
+        this._onJoin = this._onJoin.bind(this);
+    }
+
+    /**
+     * Render function of component.
+     *
+     * @return {ReactElement}
+     */
+    render() {
+        return (
+            <AtlasKitThemeProvider mode='light'>
```

Comment:
why don't use the theme provider in the root component instead of directly in the welcome page? this way, whatever other routes we add would use the same theme.

Reply:
We planned on using the dark theme in the previous meeting. So I did this.

jitsi-meet also does this. It uses light theme for welcome page only. If we use light for whole app. It shows white background during loading of conference. Dark theme makes it look good.

https://github.com/jitsi/jitsi-meet/blob/ce69ee60ca6b1eb6210e38bc3cc68cee60d42b15/react/features/app/components/App.web.js#L53

https://github.com/jitsi/jitsi-meet/blob/ce69ee60ca6b1eb6210e38bc3cc68cee60d42b15/react/features/welcome/components/WelcomePage.web.js#L106

Reply:
Got it, makes sense!